Add .editorconfig file

Sergio0694 opened the pull request:
### Description

This PR ports the .editorconfig file from the .NET Community Toolkit (https://github.com/CommunityToolkit/dotnet).

> **Note**: just a draft for now because this doesn't seem to be working, for whatever reason. If I open the solution and build and intentionally write code that doesn't respect the .editorconfig file (eg. a local using `var`), I get no warnings at all 🤔

@nirmal4g this seems like your cup of team, any ideas what might be wrong with this? Could use some help 😄

Nirmal4G reviewed (commented):
> If I open the solution and build and intentionally write code that doesn't respect the `.editorconfig` file (eg. a local using `var`), I get no warnings at all 🤔

Could be just a cache issue on Compiler side. I don't have the repo cloned. And I'm replying from mobile. I'll take a look at it soon.

Sergio0694 commented:
I don't think it's VS caching, as I've also tried on different machines and VS versions, and I never got the warnings. I'm not sure what's different than in the Toolkit repo 🤔

Nirmal4G commented on Directory.Build.props:
```diff
@@ -6,6 +6,7 @@
     <Features>strict</Features>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <NoWarn>$(NoWarn);MSB3270</NoWarn>
+    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
```
When you said warnings, I thought you can't get EditorConfig warnings but only when I cloned the repo last night and tried to build it, there were lots of errors related to CodeStyle analyzers similar to the dotnet/roslyn#63780 issue. Another issue is the incorrect placement of XAML files in the EditorConfig.

Sergio0694 replied:
> "This only works for some of the rules (specifically IDEXXX) which are above the Recommended category. See Enable Code Style Analysis on Build."

Isn't that still something we want? Like, that property is set in the .NET Community Toolkit repo, no? 🤔

Nirmal4G commented:
So, there are 3 issues at play here:

1. XAML files placed inside C# files, leading to dotnet rules not applying to C# files.

2. Code style analyzer does not match the compiler's version leading to `S.C.I` version mismatch.

3. An unrelated issue but a warning for missing Appx bundling platforms appears for a sample UWP project.

This patch fixes all of that: 18e2a7e57a8dae45c1bd52f63cc771b167cb0870

But I can't push to the PR. You can get my patch from my fork...

```cmd
git switch dev/editorconfig
git pull --ff-only https://github.com/Nirmal4G/ComputeSharp dev/editorconfig
git push
```

Nirmal4G reviewed (changes requested):
So, there are 3 issues at play here:

1. XAML files placed inside C# files, leading to dotnet rules not applying to C# files.

2. Code style analyzer does not match the compiler's version leading to `S.C.I` version mismatch.

3. An unrelated issue but a warning for missing Appx bundling platforms appears for a sample UWP project.

This patch fixes all of that: 18e2a7e57a8dae45c1bd52f63cc771b167cb0870

But I can't push to the PR. You can get my patch from my fork...

```cmd
git switch dev/editorconfig
git pull --ff-only https://github.com/Nirmal4G/ComputeSharp dev/editorconfig
git push
```

Sergio0694 commented:
Ooh, thank you! Yeah using the explicit Roslyn version is annoying, unfortunately we're stuck with that until VS 17.4 is out 🥲
Thank you for spotting that issue with .editorconfig too! I'll cherry-pick your commit and try this out again 😄

Nirmal4G commented on .editorconfig:
```diff
@@ -15,8 +15,9 @@ indent_size = 4
 
 # New line preferences
-end_of_line = unset
+end_of_line = clrf
```
But if you want to force new-line type regardless of git checkout, we can use git attributes. Just copy the file from the dotnet toolkit repo. It has defaults configured properly.

Sergio0694 replied:
Fixed, thanks! I also re-run `dotnet format whitespace --folder=ComputeSharp`, everything seems ok 🙂

Nirmal4G commented:
> In lots of files, if I open them in VS, I do see some warnings (e.g. I have some `var` uses here and there, etc.). But, if I build the whole solution from VS, I get no warnings/errors. I don't understand why those errors would only show up in IntelliSense but not in the build output? Is there a way to enforce those too?

It might be due to differences in MSBuild build, VS DesignTime build and a VS regular build. Yes, they differ subtly. I'll look into that.

Youssef1313 commented on .editorconfig:
```diff
@@ -0,0 +1,388 @@
+indent_style = space
+
+# New line preferences
+end_of_line = crlf
```
See https://github.com/dotnet/format/issues/1099, which suggests not specifying end_of_line.

I have had issues previously with `dotnet format whitespace --verify-no-changes` when running in Linux vs Windows due to that.

Nirmal4G replied:
You can remove it. No issues. As I said Git handles it automatically if you're using Git config defaults.

Sergio0694 replied:
Alright, removed in 0be0c7d4bdb649bf732e066fbb7e9f1ec49f9eef 🙂

Youssef1313 commented on .editorconfig:
```diff
@@ -0,0 +1,388 @@
+
+#### C# Coding Conventions ####
+
+# var preferences
+csharp_style_var_elsewhere = false:warning
+csharp_style_var_for_built_in_types = false:warning
+csharp_style_var_when_type_is_apparent = false:warning
```
Are you expecting these to fail CI but they are not?

This is by-design. The compiler doesn't know about the `option_key = option_value:severity` syntax.
You have to do `dotnet_diagnostic.IDExxxx.severity = warning`
`var` preferences are IDE0007 and IDE0008

Sergio0694 replied:
Resolving this as that MSBuild property to enable all IDE warnings did fix this 🙌

Youssef1313 commented on Directory.Build.props:
```diff
@@ -3,13 +3,42 @@
+      Enable the latest warning wave, which shows additional warnings for invalid language features that are disabled by default.
+      For additional info, see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves.
+    -->
+    <AnalysisLevel>7</AnalysisLevel>
```
You could try something like `7-recommended` or `7-all` which should import globalconfig from the CodeStyle package:

![image](https://user-images.githubusercontent.com/31348972/195933091-72d94559-2431-44a5-9b36-25aa56583415.png)

Sergio0694 replied:
That seems to work, but now I'm also getting a bazillion warnings for things that I actually don't want. If I want to override some of these defaults that are imported from the CodeStyle package, can I just do that in my .editorconfig?

Youssef1313 replied:
Yes. You can either enable every rule individually without using any globalconfig from the package, or use a globalconfig and start disabling what you don't need.

Sergio0694 replied:
Got it, that works! Question: what's the best way to disable a set of warnings in my "src\TerraFX.*" folders? Do I just add an .editorconfig in those folders that's not marked as root, and override the IDExxxx settings I want? 🤔

Rationale: those files are forked from another repo, so I don't want my code style to apply there as well.

Youssef1313 replied:
> Do I just add an .editorconfig in those folders that's not marked as root, and override the IDExxxx settings I want?

I was expecting this to work 😕

Sergio0694 replied:
Resolving this as we figured it out on Discord (missing `[*.cs]` headers) 😄

Youssef1313 commented on Directory.Build.props:
```diff
@@ -3,13 +3,42 @@
+      Enable the latest warning wave, which shows additional warnings for invalid language features that are disabled by default.
+      For additional info, see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves.
+    -->
+    <AnalysisLevel>7</AnalysisLevel>
```
If you want this to affect IDExxxx only, you can use:

```xml
<AnalysisLevel>7</AnalysisLevel>
<AnalysisLevelStyle>7-all</AnalysisLevelStyle>
```

Sergio0694 commented:
@Youssef1313 I've updated my .editorconfig with some better rules (eg. I fixed the fields naming convention), but if I run `dotnet format` now I also get a bunch of errors about it not being able to load some projects correctly or not being able to apply some analyzers, and as a result I only get the fixes in some files (eg. 170 something, locally), and not in all of them. I'm sure there's other bits in files that haven't been loaded that it's missing. I guess this might be related to what @Nirmal4G said about `dotnet format` not working well on projects that need MSBuild? Is there a way around this issue? 🤔

Youssef1313 commented:
@Sergio0694 What errors about "loading some projects correctly" do you see?

Sergio0694 commented:
@Youssef1313 I get all of this 😥

```
C:\Users\sergiopedri\GitHub\ComputeSharp>dotnet format ComputeSharp.sln --no-restore
Warnings were encountered while loading the workspace. Set the verbosity option to the 'diagnostic' level to log warnings.
Required references did not load for ComputeSharp.Uwp or referenced project. Run `dotnet restore` prior to formatting.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix CSharpAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicAddImportCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix GenerateTypeCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicFullyQualifyCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1007. Code fix VisualBasicSpellCheckCodeFixProvider doesn't support Fix All in Solution.
Required references did not load for ComputeSharp.Uwp or referenced project. Run `dotnet restore` prior to formatting.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
Unable to fix IDE1006. Code fix NamingStyleCodeFixProvider doesn't support Fix All in Solution.
```

Sergio0694 commented:
I've now enabled all analyzers (just disabled a few I didn't like), and fixed **all** code to produce 0 warnings 🎉

@Youssef1313 if you take another look let me know if everything looks fine to you! 😄

Youssef1313 commented on .editorconfig:
```diff
@@ -0,0 +1,383 @@
+dotnet_diagnostic.CA2241.severity = warning
+dotnet_diagnostic.CA2242.severity = warning
+
+# StyleCop Code Analysis
```
Where is the package referenced?

Sergio0694 replied:
I thought you said it's automatically used when adding `<AnalysisLevelStyle>7-all</AnalysisLevelStyle>`? Adding these suppressions here does seem to work as expected to hide them from the ones I get when building the solution 🤔

Youssef1313 replied:
Setting `AnalysisLevelStyle` to `7-all` will make IDExxxx diagnostics as warnings (this comes from `CodeStyle` package).

But StyleCop.Analyzers (holding SAxxxx diagnostics) is a separate NuGet package.

I'm not sure how you got any SAxxxx diagnostics. Probably they ship an extension that you installed in your local VS?

Sergio0694 replied:
Mmh... I guess it's very possible these SAxxxx are just leftovers and not doing anything then 😅

Youssef1313 replied:
I think they originally came from `CommunityToolkit/WindowsCommunityToolkit` repo which has:

>         <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="All" />

You could follow-up in another PR to add this package if you want.

Sergio0694 replied:
Ooh, I see. Yeah this should be fine for now, can always add that separately if needed. Thank you! 🙂

Youssef1313 commented on .editorconfig:
```diff
@@ -0,0 +1,383 @@
+# Closing parenthesis should be spaced correctly: "foo()!"
+dotnet_diagnostic.SA1009.severity = none
+
+# Hide warnings when using the new() expression from C# 9.
+dotnet_diagnostic.SA1000.severity = none
```
This was fixed in StyleCop 1.2.

See https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3561

Sergio0694 replied:
Ah, cool, thanks! Removed in https://github.com/Sergio0694/ComputeSharp/pull/404/commits/026e6717ab6a71e4577d2537cad96f1c700df1b0 🙂
